execution/src/windows/execution.cc

```diff
@@ -117,11 +118,11 @@ namespace leatherman { namespace execution {
 
         // Format a name for the pipe based on the process and counter
         wstring name = boost::nowide::widen((boost::format("\\\\.\\Pipe\\leatherman.%1%.%2%") %
-            GetCurrentProcessId() %
+            GetCurrentThreadId() %
```

Comment:
I figured someone else must have written code like this, and interestingly chromium uses both process and thread id and appends a secure random number. https://chromium.googlesource.com/chromium/src.git/+/master/base/sync_socket_win.cc#46 I imagine the pipe name is predictable otherwise.

Ruby on the other hand uses process id and a serial number (which increments in a non-thread-safeway): https://github.com/ruby/ruby/blob/trunk/win32/win32.c#L6248-L6249. ruby--